app/demo-loop/LDemoRun.cc

```diff
@@ -114,20 +114,19 @@ LDemoResult run_gpu(LDemoArgs args)
     // Load all the problem data
     LDemoParams params = load_params(args);
 
-    // Create param interfaces (TODO unify with sim/TrackInterface )
+    // Create param interfaces (TODO unify with sim/TrackInterface)
     ParamsDeviceRef params_ref = build_params_refs<MemSpace::device>(params);
 
     // Create states (TODO state store?)
     StateData<Ownership::value, MemSpace::device> state_storage;
+    // TODO: allocate correct size from LDemoParams
     resize(&state_storage,
            build_params_refs<MemSpace::host>(params),
-           args.num_tracks);
+           size_type{1});
```

Comment:
@amandalund we might need to have a "capacity" type variable for the states.... 🤔 or implement thread masking so that we initialize any empty tracks so the kernels skip over them.

Reply:
Yeah, at the moment all tracks are initialized with `alive = false` in the sim state, and I check this and exit the kernel if the track isn't alive.